### DISCUSSION TITLE
How to enable `pyright` type checking in VS Code via Pylance extension?

### DISCUSSION THREAD

**Discussion by eaftan** [microsoft/pyright#1975] [Q&A]
The README for this project says:

> For most VS Code users, we recommend using the Pylance extension rather than Pyright. Pylance incorporates the pyright type checker but features additional capabilities such as IntelliCode and semantic token highlighting. 

However, with Pylance installed and a `pyrightconfig.json` file at the root of my project directory, I do not see any type errors in VS Code.  How do I enable type checking in Pylance, preferably on a per-project basis, and preferably for anyone else who clones my project and uses VS Code (i.e., no per-user configuration)?

**Comment by jakebailey** [comment]
By default, type checking is off in Pylance (as it's noise for most users). You can either set `"python.analysis.typeCheckingMode": "basic"` (pyright's default) in a `.vscode/settings.json` file, or change the `typeCheckingMode` in the `pyrightconfig.json` file explicitly to `"basic"`.

**Reply by jakebailey** [reply]
I'd probably recommend the latter solution, as that will also allow you to have consistent settings with the pyright CLI.

**Reply by eaftan** [reply]
Thank you for the suggestion.  I set `"typeCheckingMode": "basic"` in my `pyrightconfig.json` file, and now I am getting type errors in VS Code, but I'm also getting type errors for files that I have excluded in my `pyrightconfig.json`.  How do I get Pylance to exclude those files from type checking?

**Reply by bschnurr** [reply]
r you in open files only mode?

**Reply by jakebailey** [reply]
You shouldn't be getting diagnostics from those files, unless you open them (in which case we will analyze them). Is that what you are seeing?

**Reply by eaftan** [reply]
> You shouldn't be getting diagnostics from those files, unless you open them (in which case we will analyze them). Is that what you are seeing?

Yes, that's exactly what I'm seeing -- `pyright` diagnostics from open files whether or not they're on the exclude list.  Is there some way to turn that off?

Maybe I should step up a level and explain what I'm trying to accomplish.  I am enabling type checking for an existing project.  Some of the files do not currently pass type checking.  I would like to exclude those files to start; we'll fix them 
incrementally and burn down the exclude list.  In addition, by excluding only currently failing files, any new files we create will automatically be type checked.

In addition, multiple people work on this project, and I would like (a) for them not to have to change local settings to get type checking in VS Code, and (b) for that type checking to be consistent with type checking on the CI (which I'm running via the `pyright` command line).

Is there some better way to accomplish what I'm trying to do?

**Reply by jakebailey** [reply]
`exclude` excludes files from the workspace scan (e.g. "ignore this unrelated folder that isn't really my code"); maybe what you're looking for is `ignore`.

**Reply by bschnurr** [reply]
Also, you can tag individual files for type checking with `# pyright: strict` as an alternative approach to incremental typing.

**Reply by eaftan** [reply]
`ignore` worked, that was exactly what I was looking for.  Thank you!
